ginivision/src/main/java/net/gini/android/vision/Document.java

```diff
@@ -1,58 +1,45 @@
 package net.gini.android.vision;
 
-import android.os.Parcel;
+import android.content.Intent;
 import android.os.Parcelable;
 import android.support.annotation.NonNull;
-
-import net.gini.android.vision.internal.camera.photo.ImageCache;
-import net.gini.android.vision.internal.camera.photo.Photo;
+import android.support.annotation.Nullable;
 
 
 /**
  * <p>
- * This class is the container for transferring the image of a document as a JPEG between the client application and
+ * This class is the container for transferring the image of a document as a JPEG between the client
+ * application and
  * the
  * Gini Vision Library and between the Fragments of the Gini Vision Library.
  * </p>
  *
  * <p>
- * Due to the size limitations of the {@link android.os.Bundle}, the JPEG has to be stored in a memory cache when
+ * Due to the size limitations of the {@link android.os.Bundle}, the JPEG has to be stored in a
+ * memory cache when
  * parceling and read from the cache when unparceling.
  * </p>
  *
  * <p>
- * <b>Warning:</b> Always retrieve the {@link Document} extras from a Bundle to force unparceling and removing of the
+ * <b>Warning:</b> Always retrieve the {@link Document} extras from a Bundle to force unparceling
+ * and removing of the
  * reference to
  * the JPEG byte array from the memory cache. Failing to do so will lead to memory leaks.
  * </p>
  */
-public class Document implements Parcelable {
-
-    private final byte[] mJpeg;
-    private final int mRotationForDisplay;
-
-    /**
-     * @exclude
-     */
-    public static Document fromPhoto(@NonNull Photo photo) {
-        return new Document(photo.getJpeg(), photo.getRotationForDisplay());
-    }
-
-    private Document(@NonNull byte[] jpeg, int rotationForDisplay) {
-        mJpeg = jpeg;
-        mRotationForDisplay = rotationForDisplay;
-    }
+public interface Document extends Parcelable {
```

Comment:
Refactored it into an interface for more flexibility and to separate api from implementation.

`GiniVisionDocument` contains the basic implementation which is not part of the documented public api and `ImageDocument` and `PdfDocument` the more specific implementation which are public.

For now to keep the changes for our clients minimal we should keep using the `Document` as the name, but in the future we should rename this into `GiniVisionDocument` (the implementation will then be `GiniVisionDocumentImpl`).